### PR TITLE
 SSL connection support

### PR DESCRIPTION
**Pull request by ryiwamoto**
This pull request adds support for SSL connection.

The Example of SSL connection setting is like this:

``` scala
// in Config.scala 
val sslConnection = Some(SSLSocketFactory.getDefault())
```

See https://code.google.com/p/pircbotx/wiki/Features#SSL_Support for more examples of SocketFactory.

### CODE REVIEW

File: src/main/scala/net/mtgto/irc/DefaultClient.scala [hunk 1]
```diff
@@ -78,12 +80,12 @@ class DefaultClient[T <: PircBotX](val setting: Config) extends ListenerAdapter[
...
-    setting.password match {
-      case Some(password) =>
-        innerClient.connect(setting.hostname, setting.port, password)
-      case None =>
-        innerClient.connect(setting.hostname, setting.port)
-    }
+    innerClient.connect(
+      setting.hostname,
+      setting.port,
+      setting.password.getOrElse(null),
```
Comment (mtgto) [inline src/main/scala/net/mtgto/irc/DefaultClient.scala:86]: Avoid to use null in Scala.
`setting.password.orNull` is better.

File: src/main/scala/net/mtgto/irc/Config.scala [hunk 2]
```diff
@@ -21,6 +23,7 @@ trait Config {
   // IRC Server
   val hostname: String
   val port: Int
+  val sslConnection: Option[SocketFactory]
```
Comment (mtgto) [inline src/main/scala/net/mtgto/irc/Config.scala:26]: I prefer the user can use default value like:

``` scala
val sslConnection: Option[SocketFactory] = None
```

Then the user doesn't need to write the config of sslConnection.

**Comment by mtgto** [comment]
Thank your aid. It is good idea adding SSL connection support.

**Comment by ryiwamoto** [comment]
I'm very sorry to reply late. I will look again if I have time.

**Pull request closed without merging**